tests/load/c/ginkgo_test.go

```diff
@@ -98,9 +119,38 @@ var _ = ginkgo.Describe("[Load Simulator]", ginkgo.Ordered, func() {
 			maxTPS:    60,
 			step:      5,
 		}
-		err = execute(ctx, network.PreFundedKeys, config)
+		err = execute(ctx, network.PreFundedKeys, config, metrics, logger)
 		if err != nil {
 			ginkgo.GinkgoT().Error(err)
 		}
 	})
 })
+
+// setupMetricsServer creates Prometheus server with a dynamically allocated port
+func setupMetricsServer(tc *e2e.GinkgoTestContext, registry *prometheus.Registry, networkUUID, networkOwner string, logger logging.Logger) func() {
+	r := require.New(tc)
+
+	// Allocate a port dynamically
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	r.NoError(err, "allocating dynamic port")
+
+	tcpAddr, ok := listener.Addr().(*net.TCPAddr)
+	r.True(ok, "allocating dynamic port")
+	port := tcpAddr.Port
+	r.NoError(listener.Close(), "closing listener on port %d", port)
+
+	metricsURI := fmt.Sprintf("127.0.0.1:%d", port)
+
+	metricsServer := load.NewPrometheusServer(metricsURI, registry, logger)
+	_, err = metricsServer.Start()
+	r.NoError(err, "failed to start metrics server")
+
+	monitoringConfigFilePath, err := metricsServer.GenerateMonitoringConfig(networkUUID, networkOwner)
+	r.NoError(err, "failed to generate monitoring config file")
+
+	// Return a cleanup function
+	return func() {
+		r.NoError(metricsServer.Stop(), "stopping metrics server")
```

Comment:
You can assign `metricsErrCh, err := metricsServer.Start()` and then listen to it in the cleanup

```suggestion
		select {
		case err := <-metricsErrCh:
			r.NoError(err, "metrics server error")
		default:
			r.NoError(metricsServer.Stop(), "stopping metrics server")
		}
```

Also, you might want to just inline this `setupMetricsServer` body directly in BeforeAll, since this function is getting a bit small

Reply:
Good insight, changed: https://github.com/ava-labs/avalanchego/pull/3959/commits/6dfba141e07e195d822389a6ee9e051b8b850a33